File: backend/app/schemas/conversation.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ConversationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    org_id: uuid.UUID
    contact_id: uuid.UUID
    contact_name: str
    channel: str
    status: str
    assigned_to: uuid.UUID | None
    last_message_at: datetime
    unread_count: int
    last_message_preview: str | None


class ConversationListResponse(BaseModel):
    conversations: list[ConversationResponse]

File: frontend/src/types/conversation.ts
export type Channel = "webchat" | "whatsapp" | "voice";
export type ConversationStatus = "active" | "escalated" | "resolved";

export interface Conversation {
  id: string;
  org_id: string;
  contact_id: string;
  contact_name: string;
  channel: Channel;
  status: ConversationStatus;
  assigned_to: string | null;
  last_message_at: string;
  unread_count: number;
  last_message_preview: string | null;
}
